pages/new-structure/components/navigation-bar.ts

```diff
@@ -0,0 +1,227 @@
+import { expect, type Locator, type Page } from "@playwright/test";
+
+/**
+ * Represents the navigation bar component.
+ */
+export class NavigationBar {
+  readonly page: Page;
+  /*
+   * logo Czechitas, link to home page
+   */
+  readonly logo: Locator;
+  // Domu
+  readonly home: Locator;
+  readonly forParents: Locator;
+  readonly tutorialsForParents: Locator;
+  readonly createRegistrationForParrent: Locator;
+
+  readonly forTearchers: Locator;
+  readonly tutorialsForTeachers: Locator;
+  readonly createRegistrationForTeachers: Locator;
+
+  readonly contact: Locator;
+  readonly signInButton: Locator;
+
+  readonly userIsLogged: Locator;
+  readonly profileButton: Locator;
+  readonly profileDropdownItem: Locator;
+  readonly logoutDropdownItem: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+
+    const nav = page.locator("nav");
+    this.logo = nav.locator("a.navbar-brand");
+    this.home = nav.locator("a.nav-link", { hasText: "Domů" });
+    this.forParents = nav.locator("a.nav-link", { hasText: "Pro rodiče" });
+    this.tutorialsForParents = nav.locator("a.dropdown-item", {
+      hasText: "Návody a formuláře",
+    });
+    this.createRegistrationForParrent = nav.locator("a.dropdown-item", {
+      hasText: "Vytvořit přihlášku",
+    });
+
+    // this is kinda typo? "Pro učitelé" instead of "Pro učitele"
+    this.forTearchers = nav.locator("a.nav-link", { hasText: "Pro učitelé" });
+    this.tutorialsForTeachers = nav.locator("a.dropdown-item", {
+      hasText: "Návody a formuláře",
+    });
+    this.createRegistrationForTeachers = nav.locator("a.dropdown-item", {
+      hasText: "Objednávka pro MŠ/ZŠ",
+    });
+
+    this.contact = nav.locator("a.nav-link", { hasText: "Kontakt" });
+    this.signInButton = nav.locator("a.nav-link", { hasText: "Přihlásit" });
+
+    // LOGGED
+    this.userIsLogged = nav.locator("a.nav-item", { hasText: "Přihlášen" });
+    this.profileButton = this.userIsLogged.locator("a");
+    this.profileDropdownItem = nav.locator("a.dropdown-item", { hasText: "Profil" });
+    this.logoutDropdownItem = nav.locator("a.dropdown-item", { hasText: "Odhlásit" });
+  }
+
+  /**
+   * Clicks on the logo.
+   */
+  async clickLogo() {
+    await this.logo.click();
+  }
+
+  /**
+   * Clicks on the home link.
+   */
+  async clickHome() {
+    await this.home.click();
+  }
+
+  /**
+   * Clicks on the "For Parents" link.
+   */
+  async clickForParents() {
```

Comment:
I don't view this as low-code variant. It has been debated on various meetups that these "shortcut" methods are only more code and no better readability. But I feel we can teach this as we want.

Reply:
Hmm... yeah, we can teach both. With and without.